### PR TITLE
Remove service loader based default implementation, this is meant to …

### PR DESCRIPTION
**Pull request by johnament**
…be implementation specific. (#21)

Signed-off-by: John Ament <john.d.ament@gmail.com>

@andymc12 as well

### CODE REVIEW

File: api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java [hunk 1]
```diff
@@ -41,14 +38,15 @@
  * This class provides a default implementation which uses the service loader
  * pattern to look for all implementations of <code>RestClientBuilder</code> and
  * creates a new builder with the highest priority specified by the
- * {@link Priority} annotation.
+ * {@link javax.annotation.Priority} annotation.
```
Comment (johnament) [inline api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java]: The javadoc's wrong now.

Comment (OndrejM) [reply api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java]: Yes, whole paragraph starting "This class provides..." should be removed

File: api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java [hunk 2]
```diff
@@ -58,50 +56,20 @@ protected RestClientBuilderResolver() {
     /**
      * Creates a new RestClientBuilder instance.
      * <p>
-     * Implementations are expected to override the {@link newBuilder()} method
+     * Implementations are expected to override the {@link #newBuilder()} method
      * to create custom RestClientBuilder implementations.
      * <p>
      * The default implementation uses the service loader pattern to look for
```
Comment (OndrejM) [inline api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java]: Whole paragraph about default implementation should be removed.

File: api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java [hunk 3]
```diff
@@ -117,12 +85,8 @@ public static RestClientBuilderResolver instance() {
...
                 }
 
-                ClassLoader cl = AccessController.doPrivileged(new PrivilegedAction<ClassLoader>() {
-                    @Override
-                    public ClassLoader run() {
-                        return Thread.currentThread().getContextClassLoader();
-                    }
-                });
+                ClassLoader cl = AccessController.doPrivileged(
+                    (PrivilegedAction<ClassLoader>) () -> Thread.currentThread().getContextClassLoader());
```
Comment (OndrejM) [inline api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java]: My IDE also recommended this but I'm not sure it's more readable :) 

It's probably nicer to assign the lambda to a local variable, like this:
```
PrivilegedAction<ClassLoader> action = () -> Thread.currentThread().getContextClassLoader();
ClassLoader cl = AccessController.doPrivileged(action);
```

Comment (johnament) [reply api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java]: Agreed

File: api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java (hunk 1 above)
Comment (OndrejM) [reply api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java]: I think you missed this one, @johnament 
I went on and fixed the javadocs directly with the commit https://github.com/eclipse/microprofile-rest-client/commit/d8aa302594fb2d871b03407fe67b40e29ebdf11c

**Pull request merged**